codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -53,37 +54,48 @@ protected void encode(ChannelHandlerContext ctx, MqttMessage msg, List<Object> o
      * @param message MQTT message to encode
      * @return ByteBuf with encoded bytes
      */
-    static ByteBuf doEncode(ByteBufAllocator byteBufAllocator, MqttMessage message) {
+    static ByteBuf doEncode(ChannelHandlerContext ctx,
+                     ByteBufAllocator byteBufAllocator,
+                     MqttMessage message) {
 
         switch (message.fixedHeader().messageType()) {
             case CONNECT:
-                return encodeConnectMessage(byteBufAllocator, (MqttConnectMessage) message);
+                return encodeConnectMessage(ctx, byteBufAllocator, (MqttConnectMessage) message);
 
             case CONNACK:
-                return encodeConnAckMessage(byteBufAllocator, (MqttConnAckMessage) message);
+                return encodeConnAckMessage(ctx, byteBufAllocator, (MqttConnAckMessage) message);
 
             case PUBLISH:
-                return encodePublishMessage(byteBufAllocator, (MqttPublishMessage) message);
+                return encodePublishMessage(ctx, byteBufAllocator, (MqttPublishMessage) message);
 
             case SUBSCRIBE:
-                return encodeSubscribeMessage(byteBufAllocator, (MqttSubscribeMessage) message);
+                return encodeSubscribeMessage(ctx, byteBufAllocator, (MqttSubscribeMessage) message);
 
             case UNSUBSCRIBE:
-                return encodeUnsubscribeMessage(byteBufAllocator, (MqttUnsubscribeMessage) message);
+                return encodeUnsubscribeMessage(ctx, byteBufAllocator, (MqttUnsubscribeMessage) message);
 
             case SUBACK:
-                return encodeSubAckMessage(byteBufAllocator, (MqttSubAckMessage) message);
+                return encodeSubAckMessage(ctx, byteBufAllocator, (MqttSubAckMessage) message);
 
             case UNSUBACK:
+                if (message instanceof MqttUnsubAckMessage) {
+                    return encodeUnsubAckMessage(ctx, byteBufAllocator, (MqttUnsubAckMessage) message);
+                } else {
```

Comment:
nit: you can remove the else

Reply:
Prior version calls `encodeMessageWithOnlySingleByteFixedHeaderAndMessageId` here and doesn't assume the type of `message` to be `MqttUnsubAckMessage`, therefore some client code may fail if we don't do the check here - the same way as `MqttCodecTest.testUnsubAckMessage` fails if I remove this conditional